packages/clay-popover/src/TooltipProvider.tsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import ClayPopover from './Popover';
+import React, {useEffect, useReducer, useRef} from 'react';
+import {Align} from 'metal-position';
+import {ClayPortal} from '@clayui/shared';
+
+const ALIGNMENTS = [
+	'top',
+	'top-right',
+	'right',
+	'bottom-right',
+	'bottom',
+	'bottom-left',
+	'left',
+	'top-left',
+] as const;
+
+const ALIGNMENTS_MAP = {
+	bottom: Align.Bottom,
+	'bottom-left': Align.BottomLeft,
+	'bottom-right': Align.BottomRight,
+	left: Align.Left,
+	right: Align.Right,
+	top: Align.Top,
+	'top-left': Align.TopLeft,
+	'top-right': Align.TopRight,
+};
+
+interface IState {
+	align?: typeof ALIGNMENTS[number];
+	message?: string;
+	show?: boolean;
+}
+
+const initialState: IState = {
+	align: 'top',
+	message: '',
+	show: false,
+};
+
+interface IAction extends IState {
+	type: 'align' | 'hide' | 'show';
+}
+
+const reducer = (state: IState, {type, ...payload}: IAction): IState => {
+	switch (type) {
+		case 'align':
+			return {...state, ...payload};
+		case 'show':
+			return {...state, ...payload, show: true};
+		case 'hide':
+			return {
+				...state,
+				align: initialState.align,
+				show: false,
+			};
+		default:
+			throw new TypeError();
+	}
+};
+
+const TooltipBase: React.FunctionComponent<{
+	autoAlign?: boolean;
+	children: React.ReactElement;
+	delay?: number;
+}> = ({autoAlign = true, children, delay = 600}) => {
+	const [{align, message, show}, dispatch] = useReducer(
+		reducer,
+		initialState
+	);
+
+	const timeoutIdRef = useRef<number>();
+	const targetRef = useRef<HTMLElement | null>(null);
+	const tooltipRef = useRef<HTMLElement | null>(null);
+
+	const handleHide = ({target}: any) => {
+		const dataTitle = target && target.getAttribute('data-title');
+
+		if (dataTitle) {
+			targetRef.current = null;
+
+			target.removeEventListener('click', handleHide);
+			target.setAttribute('title', dataTitle);
+			target.removeAttribute('data-title');
+
+			dispatch({type: 'hide'});
+			clearTimeout(timeoutIdRef.current);
+		}
+	};
+
+	const handleShow = ({target}: any) => {
+		const title = target && target.getAttribute('title');
+
+		if (title) {
+			targetRef.current = target;
+
+			target.addEventListener('click', handleHide);
+			target.setAttribute('data-title', title);
+			target.removeAttribute('title');
+
+			const customDelay = target.getAttribute('data-tooltip-delay');
+			const newAlign = target.getAttribute('data-tooltip-align');
+
+			timeoutIdRef.current = window.setTimeout(
```

Comment:
Probably this timeout should be around `handleShow`, it may happen that the user leaves hover on top of the element before the tooltip appears,`handleHide` will be called to remove the stuff we added but it may be unnecessary processing that we can avoid. What do you think?